core/blockchain.go

```diff
@@ -2054,6 +2063,15 @@ func (bc *BlockChain) reorg(oldBlock, newBlock *types.Block) error {
 	if err := indexesBatch.Write(); err != nil {
 		log.Crit("Failed to delete useless indexes", "err", err)
 	}
+
+	// Collect the logs
+	for i := len(newChain) - 1; i >= 1; i-- {
```

Comment:
If it's a humongous amount of blocks, maybe we shouldn't do this. If we're reorging >10K blocks, do we really want to collect and emit `3M` logs?

Reply:
I still think we maybe should limit this -- cap the logs at some reasonable number, like 1K ?